backend/Lingva.BusinessLayer/SubtitlesParser/Classes/Parsers/MicroDvdParser.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Lingva.BusinessLayer.SubtitlesParser.Classes.Parsers
+{
+    /// <summary>
+    /// Parser for MicroDVD .sub subtitles files
+    /// 
+    /// A .sub file looks like this:
+    /// {1}{1}29.970
+    /// {0}{180}PIRATES OF THE CARIBBEAN|English subtitlez by tHe.b0dY
+    /// {509}{629}Drink up me 'earties yo ho!
+    /// {635}{755}We kidnap and ravage and don't give a hoot.
+    /// 
+    /// We need the video frame rate to extract .sub files -> careful when using it
+    /// 
+    /// see https://en.wikipedia.org/wiki/MicroDVD
+    /// </summary>
+    public class MicroDvdParser : ISubtitlesParser
+    {
+        // Properties -----------------------------------------------------------------------
+
+        private readonly float defaultFrameRate = 25;
+        private readonly char[] _lineSeparators = {'|'};
+
+
+        // Constructors --------------------------------------------------------------------
+
+        public MicroDvdParser(){}
+
+        public MicroDvdParser(float defaultFrameRate)
+        {
+            this.defaultFrameRate = defaultFrameRate;
+        }
+
+
+        // Methods -------------------------------------------------------------------------
+
+        public List<SubtitleItem> ParseStream(Stream subStream, Encoding encoding)
```

Comment:
Specify please all possible exceptions at autocoments. Add It to tag   <exception cref=""></exception>